src/itensor.jl

```diff
@@ -71,11 +71,18 @@ dims(T::ITensor) = dims(inds(T))
 
 dim(T::ITensor) = dim(inds(T))
 
+Base.ndims(A::ITensor) = order(inds(A))
+size(A::ITensor) = dims(inds(A))
+size(A::ITensor, d) = d in 1:ndims(A) ? dim(inds(A)[d]) : 
+  d>0 ? 1 : error("arraysize: dimension out of range")
```

Comment:
The error type here is not consistent with `Base`

Reply:
While I was just being lazy, now that I check I see Base throws the same error here, somewhat surprisingly: 
```
@test_throws ErrorException size(rand(2),-42)
```